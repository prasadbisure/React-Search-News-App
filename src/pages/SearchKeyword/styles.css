.dropdown-container {
    width: 40%;
    margin: auto;
}

.dropdown-container ul {
    text-align: left;
    margin: 0;
    padding: 0;
    border: 1px solid #ccc;
}

.dropdown-container li {
    list-style: none;
    list-style: none;
    padding: 5px;
    border-bottom: 1px solid #ccc;
    cursor: pointer;
}