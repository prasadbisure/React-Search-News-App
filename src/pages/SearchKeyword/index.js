import './styles.css';

const SearchKeyword = ({data, handleKeywordClick}) => (
    <div className="dropdown-container">
        <ul>
            {data.map(keyword => (
                <li key={keyword.id} onClick={handleKeywordClick(keyword.webTitle)}>{keyword.webTitle}</li>
            ))}
        </ul>
    </div>
);

export default SearchKeyword;
