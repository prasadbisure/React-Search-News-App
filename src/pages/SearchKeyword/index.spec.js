import { shallow } from 'enzyme';
import SearchKeyword from './';

const props = {
    data: [ {id: 0, webTitle: 'Test'},
            {id: 1, webTitle: 'Test2'}
        ],
    handleKeywordClick: jest.fn()
}

describe('SearchKeyword component', () => {
    const wrapper =  shallow(<SearchKeyword data={props.data} handleKeywordClick={props.handleKeywordClick} />);
    
    it('should render snapshots properly', () => {
        expect(wrapper).toMatchSnapshot();
    });
})

