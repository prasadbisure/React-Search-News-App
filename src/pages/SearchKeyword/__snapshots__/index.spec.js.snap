// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchKeyword component should render snapshots properly 1`] = `
<div
  className="dropdown-container"
>
  <ul>
    <li
      key="0"
    >
      Test
    </li>
    <li
      key="1"
    >
      Test2
    </li>
  </ul>
</div>
`;
